src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
-// write code here
+const list = document.querySelector('ul');
+
+function strNum(str) {
+  const result = [];
+
+  for (let i = 1; i < str.length; i++) {
+    if (Number(str[i]) || str[i] === '0') {
+      result.push(str[i]);
+    }
+  }
+
+  return Number(result.join(''));
+}
+
+function sortList(someList) {
+  const items = Array.from(someList.getElementsByTagName('li'));
+
+  items.sort(
+    (a, b) =>
+      strNum(b.getAttribute('data-salary')) -
+      strNum(a.getAttribute('data-salary')),
+  );
+  someList.innerHTML = '';
+  items.forEach((el) => someList.appendChild(el));
+}
+
+function getEmployees(someList) {
+  const items = Array.from(someList.getElementsByTagName('li'));
+
+  return items.map((el) => ({
+    ...el,
+    'data-salary': strNum(el.getAttribute('data-salary')),
+  }));
```

Comment:
The `getEmployees` function is not returning the expected array of objects with all required attributes like `{ name, position, salary, age }`. Currently, it only returns an object with the `data-salary` attribute. Ensure that you extract and include all necessary attributes from each list item.